src/components/Fields/TextField.story.js

```diff
@@ -0,0 +1,235 @@
+    storyWrapper(
+        'Label can be forced to always appear with the hasLabelAlways prop.',
+        <TextField
+            label="Always has a label"
+            value="my example value"
+            hasLabelAlways
+        />,
+    ),
+);
+
+stories.add(
+    'Type',
+    storyWrapper(
+        'Type allows you to set the HTML input type.',
+        <TextField
+            label="With input type"
+            type="email"
+        />,
+    ),
+);
+
+stories.add(
+    'Full Width',
+    storyWrapper(
+        'isFullWidth makes an input fill the full width of it\'s container.',
+        <TextField label="Full width input" isFullWidth />,
+    ),
+);
+
+stories.add(
+    'Disabled',
+    storyWrapper(
+        'isDisabled allows you to disable the input. Disabled mode removes the line at the bottom of the input.',
+        <TextField
+            label="Disabled input"
+            value="Hello, Friend"
+            isDisabled
+        />,
+        <div>
+            <TextField
+                label="Disabled input (no value)"
+                isDisabled
+            />
+        </div>,
+    ),
+);
+
+stories.add(
+    'Read Only',
+    storyWrapper(
+        'isReadOnly allows you to make an input read only. Read only mode removes the line at the bottom of the input.',
+        <TextField
+            label="Read only input"
+            value="Hello, Friend"
+            isReadOnly
+        />,
+        <div>
+            <TextField
+                label="Read only input (no value)"
+                isReadOnly
+            />
+        </div>,
+    ),
+);
+
+stories.add(
+    'Validation',
+    storyWrapper(
+        'isValid allows you to control input validation state.',
+        <TextField label="Invalid input" isValid={false} />,
+        <div>
+            <TextField label="Valid input" isValid />
+            <TextField label="Not yet validated" isValid={null} />
+        </div>,
+    ),
+);
+
+stories.add(
+    'Tooltips',
+    storyWrapper(
+        'tooltipHint and tooltipError allow you to display tooltips to users on input focus. Error takes precedence over hint.',
+        <TextField
+            label="Input with a tooltip"
+            tooltipHint="My Example Hint"
+            tooltipError="My Example Error"
+        />,
+        <div>
+            <TextField label="Input with an error" tooltipError="My Example Error" />
+            <TextField label="Input with a hint" tooltipHint="My Example Hint" />
+            <br />
+            <TextField
+                placeholder="Required field"
+                tooltipHint="My Example Hint"
+                isRequired
+                isFullWidth
+            />
+        </div>,
+    ),
+);
+
+stories.add(
+    'Required',
+    storyWrapper(
+        'isRequired allows you to mark an input as required. And adds required icon with tooltip.',
+        <TextField
+            label="Required input"
+            value="Hello, Friend"
+            isRequired
+        />,
```

Comment:
As with `TextArea`, I think it'd be good to add `tooltipRequired` to this example because it's otherwise undocumented.